shared/src/main/java/io/pravega/shared/segment/StreamSegmentNameUtils.java

```diff
@@ -150,4 +193,114 @@ public static String getSegmentChunkName(String segmentName, long offset) {
         Preconditions.checkArgument(!segmentName.contains(OFFSET_SUFFIX), "segmentName is already a SegmentChunk name");
         return segmentName + OFFSET_SUFFIX + Long.toString(offset);
     }
+
+    /**
+     * Method to compute 64 bit segment id which takes segment number and epoch and composes it as
+     * `msb = epoch` `lsb = segmentNumber`.
+     * Primary id identifies the segment container mappeing and primary + secondary uniquely identifies a segment
+     * within a stream.
+     *
+     * @param segmentNumber segment number.
+     * @param epoch epoch in which segment was created.
+     * @return segment id which is composed using segment number and epoch.
+     */
+    public static long computeSegmentId(int segmentNumber, int epoch) {
+        Preconditions.checkArgument(segmentNumber >= 0);
+        Preconditions.checkArgument(epoch >= 0);
+        return (long) epoch << 32 | (segmentNumber & 0xFFFFFFFFL);
+    }
+
+    /**
+     * Method to extract segmentNumber from given segment id. Segment number is encoded in 32 msb of segment id
+     *
+     * @param segmentId segment id.
+     * @return segment number by extracting it from segment id.
+     */
+    public static int getSegmentNumber(long segmentId) {
```

Comment:
I find this method name confusing. In the Segment class that long that encodes both this and the epoch is called the "segmentNumber". I don't want these two methods to be mixed up.

Reply:
Either way is fine. The change so far looks good. but there are some more places I highlighted that should be updated.